plugins/categoryaslink/class.categoryaslink.plugin.php

```diff
@@ -91,9 +91,16 @@ public function categoriesController_render_before($sender, $args) {
      * @param CategoriesController $sender
      * @param CategoriesController $args
      */
-    public function categoriesController_beforeCategoriesRender_handler($sender, $args) {
+    public function categoriesController_beforeCategoriesRender_handler(CategoriesController $sender, $args) {
         if ($sender->data('Category.RedirectUrl')) {
-            redirectTo(categoryUrl($sender->Data('Category')), 301);
+            $destination = categoryUrl($sender->Data('Category'));
+            $requestRoot = Gdn::request()->getRoot();
+            // Remove $requestRoot in $destination because it will appear twice because of url() or safeURL()
+            // usage from redirectTo().
+            if ($requestRoot && strpos($destination, $requestRoot) === 0) {
+                $destination = substr($destination, strlen($requestRoot));
+            }
+            redirectTo($destination, 301);
```

Comment:
I don't think the issue is here but in the `categoryUrl()` function override just below.

The issue is that if there's a `RedirectUrl` it returns the value of `RedirectUrl` wrapped in `safeURL()` without considering Subcommunities.
Then it follows to consider Subcommunities, but only below, around line 126.

My suggestion is to leave `categoriesController_beforeCategoriesRender_handler` as is and modify `categoryUrl()` slightly to check if `class_exists('SubcommunitiesPlugin') && $category['RedirectUrl']` and then apply your logic of removing the root.

Reply:
@igraziatto 
good call. I moved my logic